invokeai/app/services/image_storage.py

```diff
@@ -207,24 +207,26 @@ def save(
         self.__set_cache(image_path, image)
         self.__set_cache(thumbnail_path, thumbnail_image)
 
-        return (image_path, thumbnail_path, int(os.path.getctime(image_path)))
+        return (image_name, thumbnail_name, int(os.path.getctime(image_path)))
 
     def delete(self, image_type: ImageType, image_name: str) -> None:
-        image_path = self.get_path(image_type, image_name)
-        thumbnail_path = self.get_path(image_type, image_name, True)
+        basename = os.path.basename(image_name)
+        image_path = self.get_path(image_type, basename)
+
         if os.path.exists(image_path):
             os.remove(image_path)
-
         if image_path in self.__cache:
             del self.__cache[image_path]
 
-        if os.path.exists(thumbnail_path):
-            os.remove(thumbnail_path)
+        thumbnail_name = f"{os.path.splitext(basename)[0]}.webp"
```

Comment:
Shouldn't you be using `get_thumbnail_name` here?

Reply:
yes i should